lib/load.php

```diff
@@ -128,3 +128,120 @@ function gutenberg_is_experiment_enabled( $name ) {
 require __DIR__ . '/block-supports/typography.php';
 require __DIR__ . '/block-supports/custom-classname.php';
 require __DIR__ . '/block-supports/border.php';
+
+
+// Demo query patterns.
+register_block_pattern_category(
+	'core/query',
+	array( 'label' => __( 'Core Query block', 'gutenberg' ) )
+);
+register_block_pattern(
```

Comment:
@kjellr can we take a look at the starting ones here? We can always tweak them at any point

Reply:
Should we add patterns in GB or in TT1-blocks first?

Reply:
I think bundling some generally-useful patterns here makes sense first (a list, a grid, some variations with and without post meta and images). I'll take some time tomorrow to look through the ones you've bundled here so far. 👍

We'll probably want to follow up this PR with some TT1-Blocks-specific patterns for that theme itself. cc @melchoyce so you can start thinking about those.

Reply:
The way it works now for Query is to have block patterns that will be used **inside Query Loop**. What this means is that we should only include patterns for a **single post** ([check existing patterns example](https://github.com/WordPress/gutenberg/pull/28891/files#diff-979d356a31350e9429b6faa5be6daaa5b10785838520396f552717a1b270c67bR143)).

If we want to support this for list/grid etc, I'll have to check it out to make it work and see how the Previews will behave :)

Reply:
Ah ok I see. The Variations show multiple posts at one time, so I figured the Patterns would work the same way: 

![Screen Shot 2021-02-11 at 9 05 41 AM](https://user-images.githubusercontent.com/1202812/107646807-630a5a00-6c48-11eb-8b86-0e3d13f54e9c.png)

This is actually one of those use cases where as a user I'm not really sure why there _are_ patterns and variations... they both do basically the same thing. For reference, here are the bundled patterns today: 

![Screen Shot 2021-02-11 at 9 14 38 AM](https://user-images.githubusercontent.com/1202812/107648089-e7111180-6c49-11eb-90a3-b736c8356cc6.png)

I was going to suggest we revise these bundled patterns to be simple, like: 

- Featured Image, Title, Excerpt stacked
- Featured Image, Title, Date stacked
- Title, Excerpt stacked
- Title, Date stacked
- (some variation of those in a grid if that's possible)

... but then I realized that half of those already exist as "Variations". If they're already covered there I'm not totally sure it makes sense to also have core provide patterns here. Maybe it only makes sense for the theme to provide its own opinionated patterns, and leave core's bundled "patterns" as "variations" instead? 

I'm curious to get other folks' thoughts on that.

Reply:
>For reference, here are the bundled patterns today:

These are just copy/paste demo patterns from me. There are no patterns for Query yet anywhere.

The difference though even with these patterns is that they contain other blocks as well for better design, like `columns`.  Existing block variations contain just some `Post Blocks` listed one after the other.

Reply:
Yeah, I see the benefit here. Here's a pass at some query block patterns we could consider bundling with Core to start with — they're a little more varied than the ones in the PR currently, and are just a little more opinionated/customized than the block variations: 

<img width="759" alt="01-full-width-image" src="https://user-images.githubusercontent.com/1202812/107815879-f6738600-6d41-11eb-8623-7cee10d20387.png">

<img width="1131" alt="02-image-on-left" src="https://user-images.githubusercontent.com/1202812/107815910-0723fc00-6d42-11eb-81a5-0b68b17142be.png">

<img width="578" alt="03-with-date" src="https://user-images.githubusercontent.com/1202812/107815882-f70c1c80-6d41-11eb-9b4a-6c14aaeda472.png">

<img width="549" alt="04-list" src="https://user-images.githubusercontent.com/1202812/107815884-f70c1c80-6d41-11eb-9736-67866c6e2800.png">

cc @melchoyce + @jasmussen for some thoughts on those. 

([Markup is available in this gist](https://gist.github.com/kjellr/a43502a5d4be823626dab250d3cd6865))

Reply:
This seems like a good start that doesn't preclude tweaks in the future. I think it's a very hard task because the patterns have to be super minimalist so as to work in any theme. I wonder if we should start even simpler, with just 3 starter patterns of varying levels of features. Such as:

- small (last one, has featured image and title only)
- medium (a variant of the 2nd one, image on the left, title, read more, perhaps no timestamp)
- big (a variant of the first one, title, wide image, timestamp, read more)

Also, we should provide some theme to these, so it doesn't read "Sample Post" and use lorem ipsum. I'll be happy to defer to you on the theme, but just throwing it out there, we reference Julio Cortázar in the Quote block description. Perhaps we could [cite some of his work](https://en.wikipedia.org/wiki/Julio_Cort%C3%A1zar#Bibliography)? Or does it have to be public domain?

Reply:
I like the small/medium/large idea. 👍 

In terms of the content, that's just what loads in from my test site. 😄 Since these are all dynamic blocks I don't think we can specify something else, and I think that's fine. 

Here's the revised set: 

Large: 
<img width="657" alt="Screen Shot 2021-02-15 at 9 55 22 AM" src="https://user-images.githubusercontent.com/1202812/107961753-5487af80-6f74-11eb-886e-9cb84f330031.png">

Medium:
<img width="682" alt="Screen Shot 2021-02-15 at 9 55 31 AM" src="https://user-images.githubusercontent.com/1202812/107961775-5782a000-6f74-11eb-9dc2-61815b06f104.png">

Small:
<img width="668" alt="Screen Shot 2021-02-15 at 9 55 39 AM" src="https://user-images.githubusercontent.com/1202812/107961784-5baebd80-6f74-11eb-8bb7-cbad36c9f45b.png">

([Updated markup](https://gist.github.com/kjellr/63d2b7acd7f4e50c7d8f17d5a04b068b))

Reply:
Oh of course, I was thinking of patterns.

There's still something to think about entirely separately, for when you load the site editor on a site without content. But yes, for another time.

Very nice!

Reply:
@kjellr I updated the patterns with a small change in `small` due to very bad spacing of `image placeholder`. I made the columns `25-75%`. We can revisit how an image placeholder is shown in really small spaces in a separate issue/PR. Is this okay?